src/ja/raw18/src/eu/kanade/tachiyomi/extension/ja/raw18/Raw18.kt

```diff
@@ -0,0 +1,127 @@
+package eu.kanade.tachiyomi.extension.ja.raw18
+
+import eu.kanade.tachiyomi.multisrc.wpcomics.WPComics
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+class Raw18 : WPComics("Raw18", "https://raw18.net", "ja", SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'", Locale.JAPANESE), null) {
+    override val searchPath = "search/manga"
+    override val genresUrlDelimiter = "="
+    override fun popularMangaSelector() = "div.items article.item"
+    override fun popularMangaNextPageSelector() = "li:nth-last-child(2) a.page-link"
+    override fun searchMangaSelector() = popularMangaSelector()
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
```

Comment:
For the whole function, use `.selectFirst()` if you only expect one matching element.